src/UI/Actions.hs

```diff
@@ -439,42 +437,42 @@ selectedItemHelper
     :: Applicative f
     => Getting (L.List n t) AppState (L.List n t)
     -> AppState
-    -> (t -> f (AppState -> AppState))
+    -> ((Int, t) -> f (AppState -> AppState))
     -> f AppState
 selectedItemHelper l s func =
   ($ s) <$> case L.listSelectedElement (view l s) of
-  Just (_, m) -> func m
+  Just m -> func m
   Nothing -> pure $ setError (GenericError "No item selected.")
 
-applyItemTags
-    :: (ListItemSetter a, Notmuch.ManageTags a, MonadIO f)
-    => a
-    -> [Text]
-    -> ([Text] -> a -> a)
-    -> AppState
-    -> f (AppState -> AppState)
-applyItemTags m ts op s =
+getEditorTagOps :: Lens' AppState (E.Editor Text Name) -> AppState -> [TagOp]
+getEditorTagOps widget s =
+  let contents = (unlines $ E.getEditContents $ view widget s)
+  in foldOf folded (parseTagOps contents)
```

Comment:
Currently this will always return an empty list in cases of a failure. I'm content with that at this point, since I'm happy to push that part of work into a separate backlog item (maybe even not for 1.0).

Reply:
True... and thanks for the *wink* :))